wire-library/wire-runtime-swift/src/main/swift/JSONString.swift

```diff
@@ -66,8 +71,35 @@ public struct JSONString<T : Hashable> : Hashable, Codable {
                 try container.encode(String(value))
             }
         default:
-            fatalError("Unsupported type \(T.self)")
+            if T.self is Optional<Int64>.Type || T.self is Optional<UInt64>.Type {
+                var container = encoder.singleValueContainer()
+                try container.encodeNil()
+            } else {
+                fatalError("Unsupported type \(T.self)")
+            }
         }
     }
 
 }
+
+private extension Decoder {
+    func decodeStringIfPresentToInt64<T>() throws -> T {
```

Comment:
These don't need to be generic anymore, right? They can just return `Int64`.

Reply:
Updated